libpng.lisp

```diff
@@ -201,7 +201,7 @@
     `(let ((,var (,(ecase direction
 			  (:input 'png-create-read-struct)
 			  (:output 'png-create-write-struct))
-		   *png-libpng-ver-string* (null-pointer)
+		   (or *png-libpng-ver-string* (setf *png-libpng-ver-string* (png-libpng-ver-string))) (null-pointer)
```

Comment:
I don't fully understand why you had to do with (other that it having to do with the time the relevant code is evaluation) … could you please explain?

Reply:
Something to do with the the function not being defined at compile time --- ie if I do (defvar *png-libpng-ver-string* (png-libpng-ver-string)) then it fails.  I could just call (png-libpng-ver-string) each time in this function, too, but seemed wasteful since it conses up a lisp string each time.  So, the short answer is no particularly good reason, I didn't want to debug what was going on.

More importantly, though, it looks like quicklisp is not pulling from master anyway --- a default quicklisp on quickload "png" grabs cl-png-vl-anyversion-04dbb706-git which is not the master branch anyway...  I can re-do this against that branch (and drop my partial implementation of the anyversion stuff).  If you merge that into master then I can just send my alpha fixes.

Reply:
I fixed it to be identical to how you did it in your anyversions branch.  I had missed the asd changes.

Reply:
Aah, I had forgotten about that branch while waiting for someone to test it. Do you mind if I merge #3 first, and then you rebase this on it? If that's easy for you, then it will allow us to get the test fixes in #3 in.

Reply:
Sounds good --- please do!

Reply:
Done!

Reply:
... and done/rebased!  The remaining change in this PR is just the alpha handling, handling channels = 2 and 4 in encode.